src/BusinessCase/JobManagement/MarathonStoreJobBusinessCase.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ *
+ * @package: chapi
+ *
+ * @author: bthapaliya
+ * @since: 2017-01-03
+ *
+ */
+
+namespace Chapi\BusinessCase\JobManagement;
+
+
+use Chapi\BusinessCase\Comparison\JobComparisonInterface;
+use Chapi\Entity\JobEntityInterface;
+use Chapi\Entity\Marathon\MarathonAppEntity;
+use Chapi\Service\JobIndex\JobIndexServiceInterface;
+use Chapi\Service\JobRepository\JobRepositoryInterface;
+use Psr\Log\LoggerInterface;
+
+class MarathonStoreJobBusinessCase extends AbstractStoreJobBusinessCase implements StoreJobBusinessCaseInterface
+{
+    public function __construct(
+        JobIndexServiceInterface $oJobIndexService,
+        JobRepositoryInterface $oJobRepositoryRemote,
+        JobRepositoryInterface $oJobRepositoryLocal,
+        JobComparisonInterface $oJobComparisonBusinessCase,
+        LoggerInterface $oLogger
+
+    )
+    {
+
+        $this->oJobIndexService = $oJobIndexService;
+        $this->oLogger = $oLogger;
+        $this->oJobComparisonBusinessCase = $oJobComparisonBusinessCase;
+        $this->oJobRepositoryRemote = $oJobRepositoryRemote;
+        $this->oJobRepositoryLocal = $oJobRepositoryLocal;
+    }
+
+    /**
+     * @return void
+     */
+    public function storeIndexedJobs()
+    {
+        $_aRemoteMissingApps = $this->oJobComparisonBusinessCase->getRemoteMissingJobs();
+        foreach ($_aRemoteMissingApps as $_sAppId)
+        {
+            $this->addRemoteMissingApp($_sAppId);
+        }
+
+        $_aLocalMissingApps = $this->oJobComparisonBusinessCase->getLocalMissingJobs();
+        foreach ($_aLocalMissingApps as $_sAppId)
+        {
+            $this->removeLocalMissingAppInRemote($_sAppId);
+        }
+        $_aLocalUpdates = $this->oJobComparisonBusinessCase->getLocalJobUpdates();
+        foreach ($_aLocalUpdates as $_sAppId)
+        {
+            $this->updateAppInRemote($_sAppId);
+        }
+    }
+
+    private function addRemoteMissingApp($sAppId)
+    {
+        if ($this->oJobIndexService->isJobInIndex($sAppId))
+        {
+            /** @var MarathonAppEntity $_oJobEntityLocal */
+            $_oJobEntityLocal = $this->oJobRepositoryLocal->getJob($sAppId);
+
+            if (!$_oJobEntityLocal instanceof MarathonAppEntity) {
```

Comment:
I hate coding style comments, but sometimes the { is in the same line. Sometimes in a new line. Is there no `php fmt` for a style guide? ;)